src/shared/sleep-config.c

```diff
@@ -32,31 +32,46 @@
 #include "strv.h"
 #include "time-util.h"
 
-int parse_sleep_config(const char *verb, bool *ret_allow, char ***ret_modes, char ***ret_states, usec_t *ret_delay) {
+SleepConfig *sleep_config_new(void) {
```

Comment:
There doesn't seem to be much need for this to be a separate function.
Just write this as:
```
sc = new0(SleepConfig, 1);
```
inline. Everything is initialized to 0 and there's no need for any assignments.

Reply:
Thanks, this makes sense.